### ISSUE TITLE
Best way to add user to image for use in Docker image

### ISSUE THREAD

**Issue by berney** [edannenberg/gentoo-bb#66]
For Docker best practices images should be configured to not run as root if they do not need the privileges, this also jives with the Least Privilege Principal. I would like to use gentoo-bb to make Docker images that do not run as root. I'm not sure of the best way to achieve this. This might seem simple to others but I'm still learning.

Extending on the figlet example from the README, in the Dockerfile I want a line like `USER figlet`, this requires that `/etc/passwd` in the image has the user figlet. As I want to create minimal images I don't want to rely on having `useradd` (or similar tools like `adduser`) in the image. I don't complete grok how gentoo-bb works yet, and if I can use tools like `adduser` at the build time inside the bob container or if I should modify the build.sh or similar.

**Comment by edannenberg** [comment]
Currently the build script just [copies](https://github.com/edannenberg/gentoo-bb/blob/master/bob-core/build-root.sh#L354) `/etc/passwd` from the build container root to the custom root as one of the last  build steps. This works around a bug/limitation in portage when using a custom `ROOT`, any users or groups created during a package install are instead created in the real root path. As we commit the build container state after each image build this is actually suprisingly clean.

tl;dr: Just run `adduser foo` in `configure_bob` or `configure_rootfs_build` hook and you should be set. You probably need to copy/create the home folder if required though.

You can also use the `install_suexec` helper function in the `finish_rootfs_build` which will you give you [su-exec](https://github.com/ncopa/su-exec) in your final image. [Example](https://github.com/edannenberg/gentoo-bb/blob/master/dock/gentoobb/images/elasticsearch/build.sh#L24).

**Comment by berney** [comment]
Thanks I see how it works now. I did as you said. install_suexec is good to know if I'll need that in an image.

I have: -

Dockerfile.template:

```
FROM ${IMAGE_PARENT}
ADD rootfs.tar /
USER figlet
CMD ["gentoo-bb"]
ENTRYPOINT ["figlet"]
```

builld.sh

```
PACKAGES="app-misc/figlet"

configure_rootfs_build() {
        useradd figlet
}
```

I didn't need a home dir, so I don't create one.
